src/aspire/classification/averager2d.py

```diff
@@ -0,0 +1,1018 @@
+        fixed_img = images[0]
+        # Difference of Gaussians (Band Filter)
+        fixed_img_dog = difference_of_gaussians(fixed_img, 1, 4)
+        # Window Images (Fix spectral boundary)
+        wfixed_img = fixed_img_dog * window("hann", fixed_img.shape)
+        # Transform image to Fourier space
+        fixed_img_fs = np.abs(fft.fftshift(fft.fft2(wfixed_img))) ** 2
+        # Compute Log Polar Transform
+        radius = fixed_img_fs.shape[0] // 8  # Low Pass
+        warped_fixed_img_fs = warp_polar(
+            fixed_img_fs,
+            radius=radius,
+            output_shape=fixed_img_fs.shape,
+            scaling="log",
+        )
+        # Only use half of FFT, because it's symmetrical
+        warped_fixed_img_fs = warped_fixed_img_fs[: fixed_img_fs.shape[0] // 2, :]
+
+        # Now prepare for rotating original images,
+        #   and searching for translations.
+        # We start back at the raw fixed_img.
+        twfixed_img = fixed_img * window("hann", fixed_img.shape)
+
+        # Register image `m` against image[0]
+        for m in range(1, len(images)):
+            # Get the image to register
+            regis_img = images[m]
+
+            # Reflect images when necessary
+            if reflection_k[m]:
+                regis_img = np.flipud(regis_img)
+
+            # Difference of Gaussians (Band Filter)
+            regis_img_dog = difference_of_gaussians(regis_img, 1, 4)
+
+            # Window Images (Fix spectral boundary)
+            wregis_img = regis_img_dog * window("hann", regis_img.shape)
+
+            self._input_images_diagnostic(
+                class_k[0], wfixed_img, class_k[m], wregis_img
+            )
+
+            # Transform image to Fourier space
+            regis_img_fs = np.abs(fft.fftshift(fft.fft2(wregis_img))) ** 2
+
+            self._windowed_psd_diagnostic(
+                class_k[0], fixed_img_fs, class_k[m], regis_img_fs
+            )
+
+            # Compute Log Polar Transform
+            warped_regis_img_fs = warp_polar(
+                regis_img_fs,
+                radius=radius,  # Low Pass
+                output_shape=fixed_img_fs.shape,
+                scaling="log",
+            )
+
+            self._log_polar_diagnostic(
+                class_k[0], warped_fixed_img_fs, class_k[m], warped_regis_img_fs
+            )
+
+            # Only use half of FFT, because it's symmetrical
+            warped_regis_img_fs = warped_regis_img_fs[: fixed_img_fs.shape[0] // 2, :]
+
+            # Compute the Cross_Correlation to estimate rotation
+            # Note that _phase_cross_correlation uses the mangnitudes (abs()),
+            #  ie it is using both freq and phase information.
+            cross_correlation, shift = self._phase_cross_correlation(
+                warped_fixed_img_fs, warped_regis_img_fs
+            )
+
+            cross_correlation_score = cross_correlation[:, 0].ravel()
```

Comment:
Had a hard time understanding what's going on here. Is it correct to say that for each angle, we're computing the sum of the correlations across radii (the zeroth Fourier component being the sum)? If so, might be worth to add a comment.

Reply:
Fair, added the following comment.

```
            # Rotating Cartesian space translates the angular log polar component.                                                                                                                                                                             
            # Scaling  Cartesian space translates the radial log polar component.                                                                                                                                                                              
            # In common image resgistration problems, both components are used                                                                                                                                                                                 
            #   to simultaneously estimate scaling and rotation.                                                                                                                                                                                               
            # Since we are not currently concerned with scaling transformation,                                                                                                                                                                                   
            #   disregard the second axis of the `cross_correlation` returned by                                                                                                                                                                               
            #   `_phase_cross_correlation`. 
  ```

Reply:
:+1: